src/main/java/com/shaft/support/DBActions.java

```diff
@@ -0,0 +1,155 @@
+		String message = "Successfully performed action [" + actionName + "].";
+		if (testData != null) {
+			message = message + " With the following test data [" + testData + "].";
+		}
+		ReportManager.log(message);
+		if (log != null) {
+			ReportManager.attachAsStep("Passed", "DB Connection Log", log);
+		}
+	}
+
+	private void passAction(String actionName, String testData) {
+		passAction(actionName, testData, null);
+	}
+
+	private void failAction(String actionName, String testData, String log) {
+		String message = "Failed to perform action [" + actionName + "].";
+		if (testData != null) {
+			message = message + " With the following test data [" + testData + "].";
+		}
+		ReportManager.log(message);
+		if (log != null) {
+			ReportManager.attachAsStep("Failed", "DB Connection Log", log);
+		}
+		Assert.fail(message);
+	}
+
+	private void failAction(String actionName, String testData) {
+		failAction(actionName, testData, null);
+	}
+
+	/**
+	 * Setup connection to database
+	 * 
+	 * @throws ClassNotFoundException
+	 * @throws SQLException
+	 */
+	public void createConnection() {
+		String testData = dbServerIP + ":" + dbPort + "/" + dbName;
+
+		try {
+			switch (dbType) {
+			// create connection to mysql DB
+			case ("mysql"):
+				Class.forName("com.mysql.jdbc.Driver");
+				connection = DriverManager.getConnection("jdbc:mysql://" + dbServerIP + ":" + dbPort + "/" + dbName,
+						username, password);
+				break;
+			// create connection to SQL Server DB
+			case ("sqlserver"):
+				Class.forName("com.microsoft.sqlserver.jdbc.SQLServerDriver");
+				connection = DriverManager.getConnection(
+						"jdbc:sqlserver://" + dbServerIP + ":" + dbPort + ";databaseName=" + dbName, username,
+						password);
+				break;
+			// create connection to SQL Server DB
+			case ("postgresql"):
+				Class.forName("org.postgresql.Driver");
+				connection = DriverManager.getConnection("jdbc:postgresql://" + dbServerIP + ":" + dbPort + "/" + dbName,
+						username, password);
+				break;
+			// create connection to oracle DB
+			case ("oracle"):
+				break;
+			}
+
+			statement = connection.createStatement();
+			passAction("create DB connection", testData);
+		} catch (SQLException e) {
+			ReportManager.log(e);
+			failAction("create DB connection", testData);
+		} catch (ClassNotFoundException e) {
+			ReportManager.log(e);
+			failAction("create DB connection", testData);
+		}
+	}
+
+	/**
+	 * execute DB query and return the result as ResultSet object
+	 * 
+	 * @param dbQuery
+	 * @return ResultSet
+	 * @throws SQLException
+	 */
+	public ResultSet executeSelectQuery(String dbQuery) {
+		try {
+			resultSet = statement.executeQuery(dbQuery);
+			passAction("Execute DB query", dbQuery);
+		}
+
+		catch (SQLException e) {
+			ReportManager.log(e);
+		}
+		return resultSet;
+	}
+
+	/**
+	 * Close database connection
+	 */
+	public void closeTheConnection() {
+		if (resultSet != null) {
+			try {
+				resultSet.close();
+			} catch (Exception e) {
```

Comment:
it is a must to never catch a generic exception object, or throwable object. if a certain exception is thrown then it should be handled. else nothing should be catched.
The same applies for the other catch statements in this class.

Reply:
Done